packages/nextjs/components/scaffold-eth/Input/AddressInput.tsx

```diff
@@ -1,28 +1,34 @@
 import { useCallback, useEffect, useState } from "react";
 import { blo } from "blo";
-import { isAddress } from "viem";
-import { Address } from "viem";
+import { useDebounce } from "usehooks-ts";
+import { Address, isAddress } from "viem";
 import { useEnsAddress, useEnsAvatar, useEnsName } from "wagmi";
-import { CommonInputProps, InputBase } from "~~/components/scaffold-eth";
-
-// ToDo:  move this function to an utility file
-const isENS = (address = "") => address.endsWith(".eth") || address.endsWith(".xyz");
+import { CommonInputProps, InputBase, isENS } from "~~/components/scaffold-eth";
 
 /**
  * Address input with ENS name resolution
  */
 export const AddressInput = ({ value, name, placeholder, onChange, disabled }: CommonInputProps<Address | string>) => {
+  // Debounce the input to keep clean RPC calls when resolving ENS names
+  // If the input is an address, we don't need to debounce it
+  const _debouncedValue = useDebounce(value, 500);
```

Comment:
hmm, how debounce helps to keep RPC calls clean? As I understand it's already clean without debounce

Reply:
I think it is because of the updated regex, since now its valid ENS once you insert `.` 

eg: it would start making RPC calls for `atg.e` , `atg.et`, and so on 

M just guessing but will try it out tomorrow 🙌

Reply:
Yep! Without the debounce, there is an RPC request on each keystroke to try and resolve an ENS name. With the debounce, the goal is to only make a request once the user is finished typing